### PR TITLE
Ben/form refactors

### PR DESCRIPTION
**Pull request by bionara**
# Description

Open to review & discussion!

Initial work on integrating universal form approach, re. validation:
* Added a FormWrapper which has `hasRequiredFields={bool}` prop to easily & uniformly set the * Required Field text - this should be used in place of any `<Form>` tags going forward
* Added required={bool} props to text, validated text, and dropdown inputs
* Added required={bool} prop to CheckList 

Implementation:
Modified CreateProfile and AboutYou (first two screen in signup flow) to:
*   use <FormWrapper hasRequiredfields={true}> 
*   disable next until form is valid
*   validate form on change/blur
* Use the required={true} on necessary inputs

## On what platform have you tested the change?

- [ ] Android - Emulator
- [ ] Android - Physical device
- [x] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots

image
image

## Checklist (for submission)

- [ ] Analytics/tracking events have been added (if needed)

## Checklist (for reviewers)

- [x] Checked against Figma designs (if relevant)
- [ ] Checked that data has been successfully saved to the backend (if relevant)

## Out of scope and potential follow-up

Address screen reader and accessibility re. required fields.
Dropdown replacement will need required={true} implementing
Possibly consider getting required={true} from the props/Yup conf, but seems overkill / passing lots of prop data vs a bool

### CODE REVIEW

File: src/components/DropdownField.tsx [hunk 1]
```diff
@@ -5,6 +5,7 @@ import { Label } from 'native-base';
 import React, { useEffect, useState } from 'react';
```
Comment (aikewoody) [inline src/components/DropdownField.tsx:5]: The Dropdownfield component is gonna be completely replaced in my branch, let's have a chat tomorrow to merge both changes

**Comment by juliatan** [comment]
Should the * be on the question text itself (at the end of the question), rather than in the placeholder?

File: src/components/DropdownField.tsx (hunk 1 above)
Comment (bionara) [reply src/components/DropdownField.tsx:5]: yep, left a note in PR description RE this cheers

**Comment by bionara** [comment]
> Should the * be on the question text itself (at the end of the question), rather than in the placeholder?

Design has it up fornt - happy to move to the back if you prefer :)

**Comment by aikewoody** [comment]
> Should the * be on the question text itself (at the end of the question), rather than in the placeholder?

Yea the asterisk should be in the question text instead of the placeholder. And I also think putting it at the end of the text is more common than at the beginning

Besides that, good work @bionara !

**Comment by juliatan** [comment]
> > Should the * be on the question text itself (at the end of the question), rather than in the placeholder?
> 
> Yea the asterisk should be in the question text instead of the placeholder. And putting it at the end of the text is more common than at the beginning
> 
> Besides that, good work @bionara !

Agree! Nice one Ben - looking forward to refactoring our forms. 😄

File: src/components/Forms.tsx [hunk 2]
```diff
@@ -15,9 +17,31 @@ export const FieldError = (props: FieldProps) => (
   </View>
 );
 
+export const requiredFormMarker = `*`;
+
+type FormWrapperProps = {
```
Comment (juliatan) [inline src/components/Forms.tsx:22]: Doesn't adhere to formal style guide 😝 `TFormWrapperProps`

Comment (bionara) [reply src/components/Forms.tsx:22]: good catch, updated!

File: src/features/multi-profile/CreateProfileScreen.tsx [hunk 3]
```diff
@@ -56,20 +57,23 @@ export default class CreateProfileScreen extends Component<RenderProps> {
...
           }}
           validationSchema={this.registerSchema}
+          validateOnChange={true}
+          validateOnBlur={true}
+          validateOnMount={true}
         >
           {(props) => {
             return (
-              <Form>
+              <FormWrapper hasRequiredFields={true}>
```
Comment (juliatan) [inline src/features/multi-profile/CreateProfileScreen.tsx:66]: Any particular reason why we don't just cut this (and other boolean attributes) down to `<FormWrapper hasRequiredFields>`?

File: src/components/Forms.tsx [hunk 4]
```diff
@@ -15,9 +17,31 @@ export const FieldError = (props: FieldProps) => (
...
+export const requiredFormMarker = `*`;
+
+type TFormWrapperProps = {
+  children: React.ReactNode;
+  hasRequiredFields?: boolean;
+};
+
+export const FormWrapper = (props: TFormWrapperProps) => (
+  <Form>
+      { props.hasRequiredFields ? (
```
Comment (aikewoody) [inline src/components/Forms.tsx:29]: I think this needs a `yarn prettier:fix`

File: src/components/Forms.tsx [hunk 5]
```diff
@@ -15,9 +17,31 @@ export const FieldError = (props: FieldProps) => (
...
+
 const styles = StyleSheet.create({
   field: {
     marginHorizontal: 16,
   },
   fieldError: {},
-});
+  requiredFieldHeader: {
+    margin: 16,
+    marginBottom: 0,
```
Comment (aikewoody) [inline src/components/Forms.tsx:45]: Do you explicitly have to set `marginBottom: 0`?

File: src/features/multi-profile/CreateProfileScreen.tsx (hunk 3 above)
Comment (aikewoody) [reply src/features/multi-profile/CreateProfileScreen.tsx:66]: Good point, it's probably something `yarn lint:fix` will do

**Review by aikewoody — DISMISSED** [review]
After running Prettier and ESLint you have my approval 👌

File: src/components/Forms.tsx (hunk 4 above)
Comment (bionara) [reply src/components/Forms.tsx]: if only i could run such a command on myself

**Review by juliatan — APPROVED** [review]

File: assets/lang/sv-SE.json [hunk 6]
```diff
@@ -1685,5 +1685,7 @@
     "title": "Din appversion behöver uppdateras!",
     "body": "Uppdatera din app för att fortsätta använda den. Vid uppdateringen får du tillgång till de senaste funktionerna.",
     "cta": "Uppdatera nu"
-  }
+  },
+
+  "required-field": "Required Field"
```
Comment (selvach) [inline assets/lang/sv-SE.json:1690]: Needs translation!

**Pull request merged**